src/shell/types.rs

```diff
@@ -193,7 +193,7 @@ impl ExecuteResult {
 }
 
 /// Reader side of a pipe.
-pub struct ShellPipeReader(os_pipe::PipeReader);
+pub struct ShellPipeReader(pub os_pipe::PipeReader);
```

Comment:
```suggestion
pub struct ShellPipeReader(os_pipe::PipeReader);
```

I'd rather not expose this dependency in the API. Is there something you needed in it?

Reply:
Going to commit this suggestion for now in order to merge this pr.

Reply:
Ah, I see why this was done now. I'm going to push some modifications.